### PR TITLE
Remove all uses of Player constructor where find is needed

### PR DESCRIPTION
**Pull request by Beagle**

### CODE REVIEW

File: deploy/lib/control/CasinoController.php [hunk 1]
```diff
@@ -20,7 +20,12 @@ class CasinoController {
 	 * @return Array
 	 */
 	public function index() {
-		$player = new Player(self_char_id());
+		$player = Player::find(self_char_id());
+
+        if (!$player) {
+            $player = new Player();
```
Comment (tchalvak) [inline deploy/lib/control/CasinoController.php]: Is the blank player so that anonymous players can view the casino?

Comment (Beagle) [inline deploy/lib/control/CasinoController.php]: yes

Comment (tchalvak) [inline deploy/lib/control/CasinoController.php]: Hmmm, shouldn't it then ideally be something like:

if(self_char_id()){
    $player ::find(self_char_id());
} else {
   $player = new Player();
}

'cause we never want there to be an existent logged in player, who can't get their player object.  If that occurs, that seems like it shouldn't overwrite with a new Player(), it should just fail fast.

Comment (Beagle) [inline deploy/lib/control/CasinoController.php]: We need a pattern for this kind of fault tolerance.

**Pull request merged**